File: Client/src/app/service/group-chat.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {GroupChatDto} from "../dto/group-chat.dto";
import { AddMessageDto } from "../dto/add-message.dto";

@Injectable({
  providedIn: 'root'
})
export class GroupChatService {

  private baseURL = "http://localhost:8080/api/groupChat";

  constructor(private http: HttpClient) {
  }

  public findAllGroupChats() {
    return this.http.get<GroupChatDto[]>(`${this.baseURL}`);
  }

  public createGroupChat(name: string) {
    return this.http.post<GroupChatDto>(`${this.baseURL}?name=${name}`, null);
  }

  public addUserToGroupChat(groupChatId: string, username: string) {
    return this.http.put(`${this.baseURL}/addUser/${groupChatId}?user=${username}`, null);
  }

  public removeUserFromGroupChat(groupChatId: string, username: string) {
    return this.http.put(`${this.baseURL}/removeUser/${groupChatId}?user=${username}`, null);
  }

  public sendMessageToGroupChat(message: AddMessageDto) {
    return this.http.post<GroupChatDto>(`${this.baseURL}/addMessage`, message);
  }
}

File: Server/src/main/java/ace/ucv/messenger/service/GroupChatService.java
package ace.ucv.messenger.service;

import ace.ucv.messenger.dto.AddMessageDto;
import ace.ucv.messenger.entity.GroupChat;

import java.security.Principal;
import java.util.List;

public interface GroupChatService {

    GroupChat createGroupChat(String name, String currentUser);

    void addUserToGroupChat(String groupId, String username);

    void removeUserFromGroupChat(String groupId, String username);

    List<GroupChat> findAllGroupChats(String username);

    GroupChat addMessage(Principal principal, AddMessageDto addMessageDto);

    void changeGroupChatName(String groupId, String newName);
}
